lib/sycamore/sycamore/docset.py

```diff
@@ -975,28 +980,48 @@ def llm_filter(
             prompt: LLM prompt.
             field: Document field to filter based on.
             threshold: Cutoff that determines whether or not to keep document.
+            ignore_doc_structure: ignores document.element mapping.
+            similarity_query: query string to compute similarity against.
+            similarity_scorer: scorer to generate similarity if 'sort_elements_by_similarity' is True.
             **resource_args
 
         Returns:
             A filtered DocSet.
         """
-
-        def threshold_filter(doc: Document, threshold) -> bool:
-            try:
-                return_value = int(doc.properties[new_field]) >= threshold
-            except Exception:
-                # accounts for llm output errors
-                return_value = False
-
-            return return_value
-
-        docset = self.filter(lambda doc: doc.field_to_value(field) is not None and doc.field_to_value(field) != "None")
-
         entity_extractor = OpenAIEntityExtractor(
             entity_name=new_field, llm=llm, use_elements=False, prompt=prompt, field=field
         )
-        docset = docset.extract_entity(entity_extractor=entity_extractor)
-        docset = docset.filter(lambda doc: threshold_filter(doc, threshold), **resource_args)
+
+        def threshold_filter(doc: Document, threshold) -> bool:
+            if ignore_doc_structure:
+                if doc.field_to_value(field) is None:
+                    return False
+                doc = entity_extractor.extract_entity(doc)
+                return int(re.findall(r"\d+", doc.properties[new_field])[0]) >= threshold
+
+            # If we're looking at a document level custom property (i.e. not text_representation),
+            # only evaluate the Document and not elements.
+            if field != "text_representation" and doc.field_to_value(field) is not None:
```

Comment:
It would be nice if we did not have `text_representation` hard-coded deep in this part of the code as meaning an element-level property that can never (?) appear at a doc level.

I suppose I would MUCH rather it be explicit when `.llm_filter()` is called whether you want to operate at the doc level or the element level. That is, rather than have something like `ignore_doc_structure`, it seems that we should have a flag `elementwise` to explicitly say we want the LLM filter to operate on the document's elements, not the top-level doc. Better yet, it seems that this should really be a separate transform that does a 'for each element do X' operation instead of baking this logic into `.llm_filter` itself. 

So for example, if I want to filter documents based on the result of an elementwise boolean operation, I could do:
```
docset.with_property('weather_is_good', lambda doc: all(doc.elementwise(lambda e: filter_weather(e))))
   .filter(lambda doc: doc.properties['weather_is_good'] == False)
   ...
```

Or something to that effect. I think we're mixing doc- and element-level operations a bit too much in the API right now and it's confusing.

My point is that it kinda feels that we're hacking in logic here for this specific case, but what we're really bumping up against is the need to perform some ops at the doc level, and others at the element level.

Reply:
>It would be nice if we did not have text_representation hard-coded deep in this part of the code as meaning an element-level property that can never (?) appear at a doc level.

That's the goal, but currently we have no way of enforcing a document model on the docset. The hope is to get there incrementally. In the current world its possible for a document object to have a text_representation and also have elements that have a text_representation, but in our logical document model we don't expect that to be the case

>I suppose I would MUCH rather it be explicit when .llm_filter() is called whether you want to operate at the doc level or the element level....

That works for me, we have a `use_elements` flag in entity extraction that basically decides whether to use the Document's information or document.elements. I don't think it's perfect but for consistency's sake maybe I just add that here? 

I dunno if we want to built the elementwise style approach yet, I think we should have a better idea of how we generally apply the data model. It seems useful but we might want to support different element projections which would impact that

Reply:
Sure, I think a decent intermediate point would be to use an "elementwise" flag or something on this function.